static/panes/cfg-view.interfaces.ts

```diff
@@ -24,8 +24,8 @@
 
 export interface CfgState {
     id: number;
-    editorid: number | boolean;
-    treeid: number | boolean;
+    editorid?: number;
+    treeid?: number;
```

Comment:
Good point about existing URLs that don't have `treeid`. If I remember correctly, the treeid can also be a `false` value from the other UI code so I think `treeid?: number | boolean` would probably be most correct.

`treeid?: number` (and same with `editorid`) would probably be better overall, not sure there's an easy way to address the source of the `false` id but they could be eliminated here https://github.com/compiler-explorer/compiler-explorer/blob/17b4f2ebc426c7d183a67a4a49b82c28640bef39/static/components.js#L158-L447

Reply:
> from the other UI code

Is this UI code already released on the site? Or just stuff that was added as part of the earlier PR? If the latter we have time to update and change that behaviour.

Reply:
> from the other UI code

Is this UI code already released on the site? Or just stuff that was added as part of the earlier PR? If the latter we have time to update and change that behaviour.

Reply:
I _think_ my recent checkin deals with this. Please take a look again! Thanks!

Reply:
From earlier code / current site behavior yeah, not sure the origin exactly!